tutorial/10_key_features/004_distributed.py

```diff
@@ -4,88 +4,306 @@
+        single_node -> many_nodes [label="Multi-node"];
+        many_nodes -> rdb_storage [label="No"];
+        many_nodes -> grpc_storage [label="Yes"];
+    }
+
+.. _multi-thread-optimization:
+
+Multi-thread Optimization
+-------------------------
+
+.. Note::
+    **Recommended backends**:
+        - :class:`~optuna.storages.InMemoryStorage`
+        - :class:`~optuna.storages.JournalStorage`
+        - :class:`~optuna.storages.RDBStorage`
+
+
+You can run multiple trials in parallel just by setting the ``n_jobs``
+parameter in :func:`~optuna.create_study()`.
+
+Multi-thread optimization has traditionally been inefficient in Python due to the Global Interpreter Lock (GIL).
+However, starting from Python 3.14 (pending official release), the GIL is expected to be removed.
+This change will make multi-threading a good option, especially for parallel optimization.
 
 """
+
+import optuna
+from optuna.storages.journal import JournalStorage, JournalFileBackend
+from optuna.trial import Trial
+import threading
+
+
+def objective_1(trial: Trial):
+    print(f"Running trial {trial.number=} in {threading.current_thread().name}")
+    x = trial.suggest_float("x", -10, 10)
+    return (x - 2) ** 2
+
+
+if __name__ == "__main__":
+    study = optuna.create_study(
+        study_name="journal_storage_multiprocess",
+        storage=JournalStorage(JournalFileBackend(file_path="./journal.log")),
+        load_if_exists=True,
+    )
+    study.optimize(objective_1, n_trials=20, n_jobs=4)
+
+
+################################################################################
+#
+# .. _multi-process-optimization:
+#
+# Multi-process Optimization with JournalStorage
+# ----------------------------------------------
+#
+# .. Note::
+#    **Recommended backends**:
+#         - :class:`~optuna.storages.JournalStorage`
+#         - :class:`~optuna.storages.RDBStorage`
+#
+# You can run multiple processes for optimization by using shared storage.
+# Since :class:`~optuna.storages.InMemoryStorage` is not designed to be shared across processes,
+# it cannot be used for multi-process optimization.
+#
+# The following example shows how to use :class:`~optuna.storages.journal.JournalStorage`
+# for multi-process optimization with ``multiprocessing`` module.
+
+import optuna
+from multiprocessing import Pool
+from optuna.storages.journal import JournalStorage, JournalFileBackend
+import os
+
+
+def objective_2(trial):
+    print(f"Running trial {trial.number=} in process {os.getpid()}")
+    x = trial.suggest_float("x", -10, 10)
+    return (x - 2) ** 2
+
+
+def run_optimization(_):
+    study = optuna.create_study(
+        study_name="journal_storage_multiprocess",
+        storage=JournalStorage(JournalFileBackend(file_path="./journal.log")),
+        load_if_exists=True,
+    )
+    study.optimize(objective_2, n_trials=3)
+
+
+if __name__ == "__main__":
+    with Pool(processes=4) as pool:
+        pool.map(run_optimization, range(12))
```

Comment:
I cannot see the output of this code. I think it is due to the double definition of `if __name__ == "__main__":` branch. How about moving them outside the branch?
```suggestion
with Pool(processes=4) as pool:
    pool.map(run_optimization, range(12))
```